test/optimization.spec.js

```diff
@@ -196,6 +196,24 @@ describe('optimization', function() {
                     return test_utils.check_optimization(start, end, type, extra);
                 });
 
+                it('optimizes reduce -every with a lot of buckets', function() {
+                    var extra = '| reduce -every :h: count()';
+                    return test_utils.check_optimization('10 years ago', 'now', type, extra);
+                });
+
+                it('optimizes reduce -every with a lot of buckets and nontrivial time filter', function() {
```

Comment:
I like this test. So much so that I'm wondering if there should be more edges testing?
Like an even end date  `var end =  2014-09-18T00:00:00.000Z;' 
And `reduce -every :day:`